staging/vhost-device-gpu/src/main.rs

```diff
@@ -55,29 +54,22 @@ impl TryFrom<GpuArgs> for GpuConfig {
 }
 
 fn start_backend(config: GpuConfig) -> Result<()> {
-
-    let handle: JoinHandle<Result<()>> = spawn(move || loop {
-        info!("Starting backend");
-        // There isn't much value in complicating code here to return an error from the threads,
-        // and so the code uses unwrap() instead. The panic on a thread won't cause trouble to the
-        // main() function and should be safe for the daemon.
-        let backend = Arc::new(RwLock::new(
-            VhostUserGpuBackend::new(config.clone()).map_err(Error::CouldNotCreateBackend)?,
-        ));
-
-        let socket = config.get_socket_path();
-
-        let mut daemon = VhostUserDaemon::new(
-            String::from("vhost-device-gpu-backend"),
-            backend,
-            GuestMemoryAtomic::new(GuestMemoryMmap::new()),
-        )
-        .map_err(Error::CouldNotCreateDaemon)?;
-
-        daemon.serve(socket).map_err(Error::ServeFailed)?;
-    });
-
-    handle.join().map_err(std::panic::resume_unwind).unwrap()
+    info!("Starting backend");
+    let backend = Arc::new(RwLock::new(
+        VhostUserGpuBackend::new(config.clone()).map_err(Error::CouldNotCreateBackend)?,
+    ));
+
+    let socket = config.get_socket_path();
+
+    let mut daemon = VhostUserDaemon::new(
+        String::from("vhost-device-gpu-backend"),
+        backend,
+        GuestMemoryAtomic::new(GuestMemoryMmap::new()),
+    )
+    .map_err(Error::CouldNotCreateDaemon)?;
+
+    daemon.serve(socket).map_err(Error::ServeFailed)?;
+    Ok(())
```

Comment:
I guess you removed the thread completely because there wasn't much use of async task if the loop was removed right?

Reply:
Well, I don't really understand why was there a thread in the first place. The thread was joined anyway. If the thread panicked the main thread would panic too (because of the `map_err(std::panic::resume_unwind)`) and this comment:
```
// There isn't much value in complicating code here to return an error from the threads,
// and so the code uses unwrap() instead. The panic on a thread won't cause trouble to the
// main() function and should be safe for the daemon.
```
Seem like it was left there possibly from a previous version (the only unwrap seem to be outside of the thread)

But maybe I should have kept it here. Not sure what is the deal with this.

Reply:
If you eventually decide whether to put back the thread spawn or not ( remember to clean up the unused imports in the main.rs)